File: src/main/java/loc/localize/generate/PhaseReplaceText.java
package loc.localize.generate;

import java.io.*;
import java.util.Map;

/**
 * Created by localadmin on 4/19/2017.
 */
public class PhaseReplaceText {

    private final String fileInput;
    private final String fileOutput;

    public PhaseReplaceText(String in, String out) {
        this.fileInput = in;
        this.fileOutput = out;
    }

    public void updateFile(Map<String, String> fileMap) throws IOException {
        BufferedReader inputStream = new BufferedReader(new FileReader(fileInput));
        File UIFile = new File(fileOutput);
        // if File doesnt exists, then create it
        if (!UIFile.exists()) {
            UIFile.createNewFile();
        }
        FileWriter filewriter = new FileWriter(UIFile.getAbsoluteFile());
        BufferedWriter outputStream = new BufferedWriter(filewriter);
        String count;
        while ((count = inputStream.readLine()) != null) {
            if (count.contains("=") && count.split("=").length > 1) {
                String tmp[] = count.split("=");
                String key = tmp[0];
                String value = tmp[1].trim().replace("\t", "");
                if (fileMap.containsKey(value)) {
                    value = fileMap.get(value);
                }
                count = key + " = " + value;
            }
            outputStream.write(count + "\n");
        }
        outputStream.flush();
        outputStream.close();
        inputStream.close();
    }
}

File: src/main/java/loc/localize/generate/PhaseReplaceText.kt
package loc.localize.generate

import java.io.*

/**
 * Created by localadmin on 4/19/2017.
 */
class PhaseReplaceText(private val fileInput: String, private val fileOutput: String) {

    @Throws(IOException::class)
    fun updateFile(fileMap: Map<String, String>) {
        val inputStream = BufferedReader(FileReader(fileInput))
        val UIFile = File(fileOutput)
        // if File doesnt exists, then create it
        if (!UIFile.exists()) {
            UIFile.createNewFile()
        }
        val filewriter = FileWriter(UIFile.absoluteFile)
        val outputStream = BufferedWriter(filewriter)
        var count: String = inputStream.readLine()
        while (count != null) {
            if (count.contains("=") && count.split("=".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray().size > 1) {
                val tmp = count.split("=".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                val key = tmp[0]
                var value = tmp[1].trim { it <= ' ' }.replace("\t", "")
                if (fileMap.containsKey(value)) {
                    value = fileMap[value].toString()
                }
                count = key + " = " + value
            }
            outputStream.write(count + "\n")
        }
        outputStream.flush()
        outputStream.close()
        inputStream.close()
    }
}
